extensions/banner/deployment/src/main/java/io/quarkus/banner/deployment/BannerProcessor.java

```diff
@@ -21,7 +21,7 @@
 
     private static final Logger logger = Logger.getLogger(BannerProcessor.class);
 
-    @BuildStep(loadsApplicationClasses = true)
+    @BuildStep(loadsApplicationClasses = true, onlyIf = IsBanner.class)
```

Comment:
question: what does that do/help ?

Reply:
ah never mind - should read the full PR brefore commenting gaf!

Reply:
Essentially the build step runs **iff** `IsBanner#getAsBoolean` returns `true`